src/ecScope.ml

```diff
@@ -2033,6 +2045,17 @@ module Theory = struct
 
       | None -> assert false
 
+  (* ------------------------------------------------------------------ *)
+  (* [external API] *)
+  let update_with_required (scope1 : scope) (scope2 : scope) =
+    let scope =
+      {scope1 with
+         sc_loaded =
+           Msym.union
+           (fun _ x y -> assert (x = y); Some x)
```

Comment:
You are structurally comparing compiled theories. This won't fly.

Reply:
See comment below.